mailpoet/lib/Automation/Integrations/WooCommerce/Fields/OrderFieldsFactory.php

```diff
@@ -0,0 +1,347 @@
+          }
+        ),
+        new Field(
+          'woocommerce:order:categories',
+          Field::TYPE_ENUM_ARRAY,
+          __('Categories', 'mailpoet'),
+          function (OrderPayload $payload) {
+            $products = $this->getProducts($payload->getOrder());
+            $categoryIds = [];
+            foreach ($products as $product) {
+              $categoryIds = array_merge($categoryIds, $product->get_category_ids());
+            }
+            sort($categoryIds);
+            return array_unique($categoryIds);
+          },
+          [
+            'options' => $this->termOptionsBuilder->getCategoryOptions(),
+          ]
+        ),
+        new Field(
+          'woocommerce:order:tags',
+          Field::TYPE_ENUM_ARRAY,
+          __('Tags', 'mailpoet'),
+          function (OrderPayload $payload) {
+            $products = $this->getProducts($payload->getOrder());
+            $tagIds = [];
+            foreach ($products as $product) {
+              $tagIds = array_merge($tagIds, $product->get_tag_ids());
+            }
+            sort($tagIds);
+            return array_unique($tagIds);
+          },
+          [
+            'options' => $this->termOptionsBuilder->getTagOptions(),
+          ]
+        ),
+        new Field(
+          'woocommerce:order:products',
+          Field::TYPE_ENUM_ARRAY,
+          __('Products', 'mailpoet'),
+          function (OrderPayload $payload) {
+            $products = $this->getProducts($payload->getOrder());
+            return array_map(function (WC_Product $product) {
+              return $product->get_id();
+            }, $products);
+          },
+          [
+            'options' => $this->getProductOptions(),
+          ]
+        ),
+      ]
+    );
+  }
+
+  private function getOrderPaymentOptions(): array {
+    $gateways = WC()->payment_gateways()->get_available_payment_gateways();
+    $options = [];
+    foreach ($gateways as $gateway) {
+      if ($gateway instanceof WC_Payment_Gateway && $gateway->enabled === 'yes') {
+        $options[] = ['id' => $gateway->id, 'name' => $gateway->title];
+      }
+    }
+    return $options;
+  }
+
+  private function getOrderStatusOptions(): array {
+    $statuses = $this->wooCommerce->wcGetOrderStatuses();
+    $options = [];
+    foreach ($statuses as $id => $name) {
+      $options[] = ['id' => $id, 'name' => $name];
+    }
+    return $options;
+  }
+
+  private function getCouponOptions(): array {
+    $coupons = $this->wordPress->getPosts([
+      'post_type' => 'shop_coupon',
+      'post_status' => 'publish',
+      'posts_per_page' => -1,
+      'orderby' => 'name',
+      'order' => 'asc',
+    ]);
+
+    $options = [];
+    foreach ($coupons as $coupon) {
+      if ($coupon instanceof WP_Post) {
+        // phpcs:ignore Squiz.NamingConventions.ValidVariableName.MemberNotCamelCaps
+        $options[] = ['id' => $coupon->post_title, 'name' => $coupon->post_name];
```

Comment:
I assume post_title is the ID because this is the code stored in the order item? But I am not sure why the `post_name` is the title and not again `post_title`?

Reply:
@websupporter Good catch, doesn't seem to make any sense to use `post_name` 👍